src/ip_notifier.cpp

```diff
@@ -55,11 +58,105 @@ namespace
 	// TODO: ip_change_notifier_sim
 #elif TORRENT_USE_NETLINK
 	// TODO: ip_change_notifier_nl
+#elif TORRENT_USE_SYSTEMCONFIGURATION
+// see https://developer.apple.com/library/content/technotes/tn1145/_index.html
+
+CFArrayRef create_keys_array()
+{
+	CFMutableArrayRef keys = CFArrayCreateMutable(nullptr
```

Comment:
As far as I can tell, ``key`` follow the "Create rule", meaning you need to call ``CFRelease()`` on it at some point